### PR TITLE
Revert all hashmap diagnostics changes

### PR DESCRIPTION
**Pull request by Manishearth**
We don't need them anymore

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18995)

### CODE REVIEW

**Comment by emilio** [comment]
r=me if it builds, hoorray!

@bors-servo r+

**Comment by emilio** [comment]
Err, this doesn't remove the begin_mutation / end_mutation calls from stylist, right?

Could we remove those too?

**Comment by Manishearth** [comment]
You missed my last commit. That was made after I checked the build.

File: components/style/custom_properties.rs [hunk 1]
```diff
@@ -168,13 +168,11 @@ where
             return;
         }
         self.index.retain(|key| !set.contains(key));
-        self.values.begin_mutation();
         // XXX It may be better to use retain when we back to use a
         // normal hashmap rather than DiagnosticHashMap.
```
Comment (emilio) [inline components/style/custom_properties.rs]: Please address this comment too, or file an issue / bug and cc Xidorn?

**Review by emilio — APPROVED** [review]

**Comment by Manishearth** [comment]
@bors-servo r-

will fix

**Pull request merged**